bundle/bundle.go

```diff
@@ -1678,6 +1680,37 @@ func readFile(f *Descriptor, sizeLimitBytes int64) (bytes.Buffer, error) {
 		return *bb, nil
 	}
 
+	// Case for *lazyFile readers:
+	if lf, ok := f.reader.(*lazyFile); ok {
+		var buf bytes.Buffer
+		if lf.file == nil {
+			var err error
+			if lf.file, err = os.Open(lf.path); err != nil {
+				return buf, fmt.Errorf("failed to open file %s: %w", f.path, err)
+			}
+		}
+		// Bail out if we can't read the whole file-- there's nothing useful we can do at that point!
+		fileSize, _ := fstatFileSize(lf.file)
+		if fileSize > sizeLimitBytes {
+			return buf, fmt.Errorf(maxSizeLimitBytesErrMsg, strings.TrimPrefix(f.Path(), "/"), fileSize, sizeLimitBytes-1)
+		}
+		// Prealloc the buffer for the file read.
+		buffer := make([]byte, fileSize)
+		_, err := lf.file.Read(buffer)
```

Comment:
Is `Read()` guaranteed to read the full file?
If not sure, `io.ReadFull()` doc claims: 

> ReadFull reads exactly len(buf) bytes from r into buf

If it has the same performance, perhaps it's a safer bet?

Reply:
For what I can tell from the docs/sources on [`File.Read`](https://pkg.go.dev/os#File.Read), both the new special case, and the original `Reader.Read` in the fallback case below this block, would have identical behavior (reading the full file if possible, but reading just the available chunks if the file descriptor was a stream).

Under-the-hood, `io.ReadFull` uses `io.ReadAtLeast`, which does `Reader.Read()` calls in a loop until either it gets an EOF, or successfully reads `len(buf)`-many bytes. I think this handles a few more special cases, at the expense of requiring more error handling on our parts. I'll give it a whirl, and see how it does.

Reply:
Checked the heap/alloc graphs, and GC reports-- I'm leaving `io.ReadFull` in, since it's closer to the *intended* behavior for `lazyFile` reading, and seems to have no major impacts on performance, despite a bit of added machinery around the actual file reading.